client/src/modules/purchases/create/createUpdate.html

```diff
@@ -124,6 +124,25 @@
                       <div ng-messages-include="modules/templates/messages.tmpl.html"></div>
                     </div>
                   </div>
+
+                  <div
+                    class="form-group"
+                    ng-class="{'has-error' : PurchaseOrderForm.$submitted && PurchaseOrderForm.shipping_handling.$invalid }">
+                    <label class="control-label">
+                      <span translate>FORM.LABELS.SHIPPING_HANDLING_COST</span> ({{PurchaseCtrl.rate.currency.symbol}})</label>
+                    <input
+                      ng-model="PurchaseCtrl.order.details.shipping_handling"
+                      name="shipping_handling"
+                      class="form-control"
+                      type="number"
+                      min="0"
+                      onkeypress="return event.charCode != 45"
```

Comment:
Why do you have this onkeypress event?

Reply:
I added this to prevent entering an minus sign, thus preventing negative values for shipping and handling.

Reply:
Why doesn't the "min" work?  It should block submission if any of the fields are incorrect.

Reply:
It does work, but only when the input element gets verified.  This prevents negative values during entry.

Reply:
In my view, that is the best solution.  Currently, someone might think their keyboard is broken, rather than thinking that the data is invalid.  If you want to validate the data (and show a red indicator during entry), just change the `ng-class` on the `<div class="form-group">` to make it constantly active.  It will light up red when the user enters invalid data, rather than silently ignoring their input.

However, pretty much all our forms work in the old way, where we define a data entry phase and then a review phase where anything wrong is marked in red.  I find it confusing if this is different for this specific form.  Is it really that much of a value add to change the behavior for this specific input?

Reply:
Ok. Removed.